question-servers/freeform.js

```diff
@@ -149,6 +151,101 @@ module.exports = {
         });
     },
 
+    loadExtensions(sourceDir, callback) {
+        async.waterfall([
+            (callback) => {
+                fs.readdir(sourceDir, (err, elements) => {
+                    if (err && err.code === 'ENOENT') {
+                        /* We don't really care if there are no extensions, just return an empty array. */
+                        return callback(null, []);
+                    }
+                    if (ERR(err, callback)) return;
+                    return callback(null, elements);
+                });
+            },
+            (elementFolders, callback) => {
+                /* Get extensions from each element folder */
+                async.map(elementFolders, (element) => {
+                    fs.readdir(path.join(sourceDir, element), (err, extensions) => {
+                        if (ERR(err, callback)) return;
+                        return callback(null, extensions.map(ext => [element, ext]));
+                    });
+                }, callback);
+            },
+            (extensions, callback) => {
+                /* Filter out non-directories */
```

Comment:
Do we need to do this, given that we are going to skip cases where we can't open `info.json` below? I'm thinking of the way we handle this when loading the questions directory, for example.

Reply:
Hmm, you're right that this is probably safe to remove.  I'll try removing this from `loadElements()` as well and see if anything breaks.

Reply:
It was unnecessary, removed.